ibis/backends/pyspark/__init__.py

```diff
@@ -1283,10 +1283,12 @@ def _to_filesystem_output(
     ) -> StreamingQuery | None:
         df = self._session.sql(self.compile(expr, params=params, limit=limit))
         if self.mode == "batch":
-            df = df.write.format(format)
+            df = df.write.format(format).mode(
+                "overwrite"
+            )  # df = df.write.format(format)
             for k, v in (options or {}).items():
                 df = df.option(k, v)
-            df.save(path)
+            df.save(os.fspath(path))  # df.save(path)
```

Comment:
@chloeh13q this is the change you suggested in #9790 but with only that change I get. I added mode overwrite but I'm not sure if that's what we want, plus the test fails with an assertion error. I think I might be missing something on how to_parquet works for pyspark.  
```
E               pyspark.errors.exceptions.captured.AnalysisException: [PATH_ALREADY_EXISTS] Path file:/private/
var/folders/ky/_zw0jkyd7hl3f9hfy6c994zw0000gn/T/pytest-of-naty/pytest-27/test_table_to_parquet_dir_pysp0
already exists. Set mode as "overwrite" to overwrite the existing path.
```
cc: @cpcloud since you asked about the test

Reply:
See my comment below, this is a bug in your test code, and we shouldn't do anything like overwrite by default.